src/user/guards/is-user-author.guard.ts

```diff
@@ -17,14 +17,17 @@ export class IsUserAuthorGuard implements CanActivate {
     const recipeFromDb = await this.recipeService.findRecipeById(
       Number(request.params.id),
     );
+    try {
+      if (request.user.id !== recipeFromDb.authorId) {
+        throw new HttpException(
```

Comment:
since this is you database, you do not have to throw an error and catch it. 
You could simply return false if user is not an author or if a recipe is not found in other case return true